From 51b970cf88a3b59382dc45f7a32d8527ecb6c63b Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Wed, 07 Feb 2024 11:43:17 +0200
Subject: [PATCH] [FSW-334] fs1: config psoc drain

Enable address stamp saving mechanism for PARC transactions.
Drain is enabled by default.

Signed-off-by: Igal Zeltser <izeltser@habana.ai>
Change-Id: Ibbcedfc9c5bffc2b6aa34f08b6a42dd73b91b0cb
---

diff --git a/arch/fs1/arc/common.c b/arch/fs1/arc/common.c
index 5fd4410..c3acd78 100644
--- a/arch/fs1/arc/common.c
+++ b/arch/fs1/arc/common.c
@@ -21,6 +21,8 @@
 #include <arc_local_conf_masks.h>
 #include <arc_regs.h>
 #include <pif_arc_regs.h>
+#include <axi_drain_wrap_regs.h>
+#include <axi_drain_wrap_masks.h>
 #include <fs1_blocks_linux_driver.h>
 
 #include <baremetal.h>
@@ -204,8 +206,21 @@
 
 __weak void hl_dev_psoc_pll_init(void) {}
 
+#define DRAIN_STAMP_ENA_MASK	(AXI_DRAIN_WRAP_RSP_ERR_DRAIN_STAMP_RD_EN_M | \
+				 AXI_DRAIN_WRAP_RSP_ERR_DRAIN_STAMP_WR_EN_M)
+
+static void hl_psoc_drain_init(void)
+{
+	/* psoc drain is enabled by default, enable drain address stamp logging */
+	regmap_writel(DRAIN_STAMP_ENA_MASK, mmIO_0_CFG_HALF_0_PSOC_PARC_CFG_MISC_DRAIN_HBW_BASE +
+					    mmAXI_DRAIN_WRAP_RSP_ERR_DRAIN_STAMP);
+	regmap_writel(DRAIN_STAMP_ENA_MASK, mmIO_0_CFG_HALF_0_PSOC_PARC_CFG_MISC_DRAIN_LBW_BASE +
+					    mmAXI_DRAIN_WRAP_RSP_ERR_DRAIN_STAMP);
+}
+
 void hl_plat_early_init(void)
 {
+	hl_psoc_drain_init();
 	hl_init_cpu_ids();
 	hl_lbw_early_write_resp_enable();
 	hl_dev_psoc_pll_init();
